### PR TITLE
cmd/lncli: always parse the pay_addr field for sendpayment

### PR DESCRIPTION
**Pull request by Roasbeef**
In this commit, we fix a bug that would cause attempts to re-use an AMP
invoice to fail. Without this commit, we would only attempt to parse the
payment addr if no invoice was specified, so a user manually specifying the
pubkey of the detonation. The fix is straight forward: always parse the
`pay_addr` field as the user may be attempting to re-use an AMP invoice w/o
open coding each of the sections.

### CODE REVIEW

**Comment by Roasbeef** [comment]
Thinking we should take it another step forward, and just either add a CLI-only flag for re-use, or one on the RPC interface. Leaning more towards the CLI flag, as adding something to the RPC interface is a bigger commitment in a sense, and after we fix the invoice DB blocker, users will be able to do do payinvoice as normal.

File: cmd/lncli/cmd_pay.go [hunk 1]
```diff
@@ -250,6 +277,16 @@ func sendPayment(ctx *cli.Context) error {
 			Amt:            ctx.Int64("amt"),
 		}
 
+		// We'll attempt to parse a payment address as well, given that
+		// if the user is uysing an AMP invoice, then they may be
```
Comment (yyforyongyu) [inline cmd/lncli/cmd_pay.go:281]:
```suggestion
		// if the user is using an AMP invoice, then they may be
```

File: cmd/lncli/cmd_pay.go [hunk 2]
```diff
@@ -250,6 +277,16 @@ func sendPayment(ctx *cli.Context) error {
...
+		// trying to specify that value manually.
+		payAddr, err := parsePayAddr(ctx)
```
Comment (yyforyongyu) [inline cmd/lncli/cmd_pay.go:283]: hmmm this is weird, not familiar with AMP yet, shouldn't the payment address be encoded in `pay_req`?

**Review by yyforyongyu — COMMENTED** [review]
> Thinking we should take it another step forward, and just either add a CLI-only flag for re-use

Like mark reuse of the address?

File: cmd/lncli/cmd_pay.go (hunk 2 above)
Comment (Roasbeef) [reply cmd/lncli/cmd_pay.go:283]: So the idea is that we're using the invoice here just to carry details such as the destination, the final CLTV to use, hop-hints, etc. With AMP, a user can specify another pay_addr, and we use that to do a normal push payment.

**Comment by Roasbeef** [comment]
> Like mark reuse of the address?

Yeah so a new flag that'll auto generate the random 32 bytes for the user, makes things a bit easier.

**Review by yyforyongyu — APPROVED** [review]
LGTM👍 Just a typo needs to be fixed.

**Comment by Roasbeef** [comment]
Fixed typo issue, tacked on the easy re-use flag also.

**Pull request merged**